src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -414,20 +421,19 @@ private BoundExpression VisitListInterfaceCollectionExpression(BoundCollectionEx
             Debug.Assert(_factory.ModuleBuilderOpt is { });
             Debug.Assert(_diagnostics.DiagnosticBag is { });
             Debug.Assert(node.Type is NamedTypeSymbol);
-            Debug.Assert(node.CollectionCreation is null);
```

Comment:
intentionally removed.  in the case of something like `with(capacity: 4)` tehre will be a CollectionCreation expression computed that calls the appropriate constructor.